cli/src/main/java/com/devonfw/tools/ide/io/FileAccess.java

```diff
@@ -131,4 +132,12 @@ default void copy(Path source, Path target) {
    */
   Path findFirst(Path dir, Predicate<Path> filter, boolean recursive);
 
+  /**
+   * Retrieves a list of file paths from the specified directory that satisfy the given predicate.
+   * @param dir the folder to iterate through
+   * @param filter the {@link Predicate} that determines whether a file should be included in the list.
+   * @return a list of paths that satisfy the provided {@link Predicate} or {@code null} if no match was found.
+   */
+  List<Path> getFilesInDir(Path dir, Predicate<Path> filter);
```

Comment:
actually this method does not only return files but also folders. It should therefore be named `getChildren` or `getChildrenOfDir`.

Reply:
done